### DISCUSSION TITLE
oauth2 authentication through backend( spring boot backend api)

### DISCUSSION THREAD

**Discussion by rahulcse1** [nextauthjs/next-auth#9349] [Help]
Hi Team i have an exisitng app written in reactjs and a backend which is in spring boot. 
My backend basically take care of oauth2 user creation, saving to database and finally returning token to frontend. 

Recently we switch to nextjs and wanted to include the next-auth. so the requirement is -
when i click on "Login with Google" button => and after authenticate with google => After that it should hit my backend where i can resume my exiting user creation and saving and finally create a token and send it back to frontend and manage my session through nextjs. 

I know i can save user in my callback but then i am not sure how to get token from my api. 

**Behaviour -

React App
1. Click "Login with Google" 
2. render google page once authenticate 
3. call goes to my backend api 
4. backend api fetch google profile and user saved in db
5. create token and send it back to FE.

Nextjs with nextauth
1. Click "Login with Google" 
2. render google page once authenticate
3. getting google profile at FE.** 

is there any way to change this behaviour ?

**Comment by pangeaos** [comment]
What was your solution about this? I have same questions about the implementation

**Comment by MoalosiLiteboho** [comment]
if you have found the approach please help .. 
i still want the approach to connect authjs with spring boot oauth2 server application
